301-jenkins-acr-spinnaker-k8s/azuredeploy.json

```diff
@@ -1,350 +1,338 @@
+            }
+        },
+        "servicePrincipalAppKey": {
+            "type": "securestring",
+            "metadata": {
+                "description": "Service Principal App Key that has contributor rights to the subscription used for this deployment. It is used by the Kubernetes cluster to dynamically manage resources (e.g. user-defined load balancers)."
+            }
+        },
+        "gitRepository": {
+            "type": "string",
+            "metadata": {
+                "description": "The URL to a public git repository used for the default Jenkins job. It must include a Dockerfile at the root of the repo."
+            },
+            "defaultValue": "https://github.com/lwander/spin-kub-demo.git"
+        },
+        "dockerRepository": {
+            "type": "string",
+            "metadata": {
+                "description": "The repository name used by the default Jenkins job and Spinnaker pipeline. This repository will be created in your Azure Container Registry."
+            },
+            "defaultValue": "lwander/spin-kub-demo"
         }
-      }
     },
-    {
-      "apiVersion": "2016-09-01",
-      "type": "Microsoft.Network/publicIPAddresses",
-      "name": "[variables('publicIPAddressName')]",
-      "location": "[resourceGroup().location]",
-      "properties": {
-        "publicIPAllocationMethod": "Dynamic",
-        "dnsSettings": {
-          "domainNameLabel": "[parameters('devopsDnsPrefix')]"
-        }
-      }
+    "variables": {
+        "resourcePrefix": "devops",
+        "storageAccountName": "[concat(variables('resourcePrefix'), uniquestring(resourceGroup().id))]",
+        "acrStorageAccountName": "[concat('registry', uniqueString(resourceGroup().id))]",
+        "acrName": "[uniqueString(resourceGroup().id)]",
+        "OSDiskName": "[concat(variables('resourcePrefix'), 'OSDisk')]",
+        "nicName": "[concat(variables('resourcePrefix'), 'VMNic')]",
+        "subnetName": "[concat(variables('resourcePrefix'), 'Subnet')]",
+        "publicIPAddressName": "[concat(variables('resourcePrefix'), 'PublicIP')]",
+        "vmStorageAccountContainerName": "vhds",
+        "vmName": "[concat(variables('resourcePrefix'), 'VM')]",
+        "virtualNetworkName": "[concat(variables('resourcePrefix'), 'VNET')]",
+        "vmExtensionName": "initializeDevOps",
+        "frontEndNSGName": "[concat(variables('resourcePrefix'), 'NSG')]",
+        "kubernetesName": "[concat('containerservice-', resourceGroup().name)]",
+        "kubernetesMasterCount": 1,
+        "kubernetesDnsPrefix": "[concat('k8s', uniquestring(resourceGroup().id))]",
+        "pipelinePort": "8000",
+        "_artifactsLocation": "https://raw.githubusercontent.com/Azure/azure-devops-utils/v0.8.0/",
```

Comment:
let's rename these since they are not used in the standard way prescribed in best practices...  maybe something like devopsUtilsLocation